jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/KModuleDeploymentService.java

```diff
@@ -253,19 +253,12 @@ protected RuntimeEnvironmentBuilder boostrapRuntimeEnvironmentBuilder(KModuleDep
     	DeploymentDescriptor descriptor = deploymentUnit.getDeploymentDescriptor();
     	if (descriptor == null || ((DeploymentDescriptorImpl)descriptor).isEmpty()) { // skip empty descriptors as its default can override settings
 	    	DeploymentDescriptorManager descriptorManager = new DeploymentDescriptorManager("org.jbpm.domain");
-            List<DeploymentDescriptor> descriptorHierarchy = DeploymentDescriptorManagerUtil.getDeploymentDescriptorHierarchy(descriptorManager, kieContainer);
-
-			descriptor = merger.merge(descriptorHierarchy, mode);
-			deploymentUnit.setDeploymentDescriptor(descriptor);
+	    	descriptor = getDeploymentDescriptor(descriptorManager, kieContainer, mode);
     	} else if (descriptor != null && !deploymentUnit.isDeployed()) {
     		DeploymentDescriptorManager descriptorManager = new DeploymentDescriptorManager("org.jbpm.domain");
-            List<DeploymentDescriptor> descriptorHierarchy = DeploymentDescriptorManagerUtil.getDeploymentDescriptorHierarchy(descriptorManager, kieContainer);
-
-	    	descriptorHierarchy.add(0, descriptor);
-	    	descriptor = merger.merge(descriptorHierarchy, mode);
-			deploymentUnit.setDeploymentDescriptor(descriptor);
+    		descriptor = getDeploymentDescriptor(descriptorManager, kieContainer, mode, deploymentUnit.getDeploymentDescriptor());
     	}
-
+        deploymentUnit.setDeploymentDescriptor(descriptor);
```

Comment:
Previously, descriptor was only set in both cases of previous "if...else.."; however, now it would be set even if deploymentUnit is deployed, is it right?

Reply:
not sure about this. But I will change it how it was previously.